### PR TITLE
Refactor `Igce` script include with fixes for Travel and Training

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
# Overview
Due to a change in JSON structure on the frontend, need to refactor script for Travel and Training.

## Current Testing (Travel and Training Only)
`ACQ0000416`

## Old Testing results
The packages have had their contents deleted, however the results of the partial tests show the mapping working as expected.

`ACQ0000399` 
```json
{
    "documentType": "INDEPENDENT_GOVERNMENT_COST_ESTIMATE",
    "templatePayload": {
        "instructions": {
            "travel_option": "SINGLE",
            "travel_estimated_values": "{\"PER_PERIOD\":200.00}",
            "surge_requirement_capabilities": "NO",
            "surge_requirement_capacity": null,
            "how_est_dev_contracting_office_other_charges_fee": null,
            "how_est_dev_contracting_office_other_fee_percentage": null,
            "how_est_dev_prev_cost_estimate_comp_option": null,
            "how_est_dev_prev_cost_estimate_comp_percentage": null,
            "how_est_dev_cost_estimate_description": null,
            "how_est_dev_tools_used": null,
            "how_est_dev_other_tools_used": null,
            "sys_id": "5978635647a861104a251bd8c26d43a8"
        },
        "periodsEstimate": [
            {
                "period": {
                    "periodType": "BASE",
                    "periodUnitCount": "1",
                    "periodUnit": "YEAR",
                    "optionOrder": "1"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "4 x (France, 2 days, 3 travelers);20 x (Canada, 10 days, 20 travelers);10 x (Japan, 10 days, 10 travelers);",
                        "unitPrice": 200,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "2",
                    "periodUnit": "MONTH",
                    "optionOrder": "2"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "4 x (France, 2 days, 3 travelers);20 x (Canada, 10 days, 20 travelers);",
                        "unitPrice": 200,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "3",
                    "periodUnit": "WEEK",
                    "optionOrder": "3"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "4 x (France, 2 days, 3 travelers);",
                        "unitPrice": 200,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "4",
                    "periodUnit": "DAY",
                    "optionOrder": "4"
                },
                "periodLineItems": []
            }
        ],
        "travel_test": [
            {
                "selected_periods": "1f98635647a861104a251bd8c26d43b2,9b98635647a861104a251bd8c26d43cc,1b98635647a861104a251bd8c26d4392",
                "duration_in_days": 2,
                "number_of_travelers": 3,
                "trip_location": "France",
                "number_of_trips": 4,
                "sys_id": "1c19275647a861104a251bd8c26d4330",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 200
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "2",
                            "periodUnit": "MONTH",
                            "optionOrder": "2",
                            "quantity": 200
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "3",
                            "periodUnit": "WEEK",
                            "optionOrder": "3",
                            "quantity": 200
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": 200,
                "travel_description": "4 x (France, 2 days, 3 travelers);"
            },
            {
                "selected_periods": "9b98635647a861104a251bd8c26d43cc,1f98635647a861104a251bd8c26d43b2",
                "duration_in_days": 10,
                "number_of_travelers": 20,
                "trip_location": "Canada",
                "number_of_trips": 20,
                "sys_id": "9419e35647a861104a251bd8c26d4308",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "2",
                            "periodUnit": "MONTH",
                            "optionOrder": "2",
                            "quantity": 200
                        },
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 200
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": 200,
                "travel_description": "20 x (Canada, 10 days, 20 travelers);"
            },
            {
                "selected_periods": "1f98635647a861104a251bd8c26d43b2",
                "duration_in_days": 10,
                "number_of_travelers": 10,
                "trip_location": "Japan",
                "number_of_trips": 10,
                "sys_id": "d0192b5647a861104a251bd8c26d43f7",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 200
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": 200,
                "travel_description": "10 x (Japan, 10 days, 10 travelers);"
            }
        ]
    }
}
```

`ACQ0000397` 

```json
{
    "documentType": "INDEPENDENT_GOVERNMENT_COST_ESTIMATE",
    "templatePayload": {
        "instructions": {
            "travel_option": "MULTIPLE",
            "travel_estimated_values": "{\"c029571a476861104a251bd8c26d4342\":5000.00,\"8029571a476861104a251bd8c26d4366\":4000.00,\"cc29d31a476861104a251bd8c26d4346\":3000.00,\"0c29d31a476861104a251bd8c26d4352\":2000.00}",
            "surge_requirement_capabilities": "YES",
            "surge_requirement_capacity": 20,
            "how_est_dev_contracting_office_other_charges_fee": null,
            "how_est_dev_contracting_office_other_fee_percentage": null,
            "how_est_dev_prev_cost_estimate_comp_option": "MORE_THAN",
            "how_est_dev_prev_cost_estimate_comp_percentage": 2,
            "how_est_dev_cost_estimate_description": "Using servicenow",
            "how_est_dev_tools_used": ",AWS",
            "how_est_dev_other_tools_used": null,
            "sys_id": "92d8d31a476861104a251bd8c26d434b"
        },
        "periodsEstimate": [
            {
                "period": {
                    "periodType": "BASE",
                    "periodUnitCount": "1",
                    "periodUnit": "YEAR",
                    "optionOrder": "1"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "refactor_asap",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #2",
                        "itemDescription": "Trying to become a Sith Lord, hoping to get promoted to Darth next year.",
                        "unitPrice": 20,
                        "quantity": 2,
                        "unit": "MONTHLY_SUBSCRIPTION"
                    },
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "43 x (The Sun, 4 days, 2 travelers);3 x (Hogwarts, 3 days, 4 travelers);3 x (Aruba, 3 days, 4 travelers);",
                        "unitPrice": 2000,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "2",
                    "periodUnit": "MONTH",
                    "optionOrder": "2"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "12 x (Atlantis, 2 days, 6 travelers);43 x (The Sun, 4 days, 2 travelers);3 x (Aruba, 3 days, 4 travelers);",
                        "unitPrice": 3000,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "3",
                    "periodUnit": "WEEK",
                    "optionOrder": "3"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "refactor_asap",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #1",
                        "itemDescription": "Need to train my force powers, trying to become a Jedi Knight in Q3.",
                        "unitPrice": 100,
                        "quantity": 3,
                        "unit": "PER_PERSON"
                    },
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "43 x (The Sun, 4 days, 2 travelers);1 x (The Moon, 10 days, 8 travelers);3 x (Aruba, 3 days, 4 travelers);",
                        "unitPrice": 4000,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "4",
                    "periodUnit": "DAY",
                    "optionOrder": "4"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "refactor_asap",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #3",
                        "itemDescription": "Need to actually lose security knowledge",
                        "unitPrice": 1000,
                        "quantity": 1,
                        "unit": "year"
                    },
                    {
                        "idiqClin": "refactor_asap",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #1",
                        "itemDescription": "Need to train my force powers, trying to become a Jedi Knight in Q3.",
                        "unitPrice": 100,
                        "quantity": 4,
                        "unit": "PER_PERSON"
                    },
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "FFP",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "3 x (Aruba, 3 days, 4 travelers);",
                        "unitPrice": 5000,
                        "quantity": 1,
                        "unit": "period"
                    }
                ]
            }
        ],
        "travel_test": [
            {
                "selected_periods": "cc29d31a476861104a251bd8c26d4346",
                "duration_in_days": 2,
                "number_of_travelers": 6,
                "trip_location": "Atlantis",
                "number_of_trips": 12,
                "sys_id": "170bd75a476861104a251bd8c26d430f",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "2",
                            "periodUnit": "MONTH",
                            "optionOrder": "2",
                            "quantity": 3000
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": [
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "4",
                        "periodUnit": "DAY",
                        "optionOrder": "4",
                        "quantity": 5000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "3",
                        "periodUnit": "WEEK",
                        "optionOrder": "3",
                        "quantity": 4000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "2",
                        "periodUnit": "MONTH",
                        "optionOrder": "2",
                        "quantity": 3000
                    },
                    {
                        "periodType": "BASE",
                        "periodUnitCount": "1",
                        "periodUnit": "YEAR",
                        "optionOrder": "1",
                        "quantity": 2000
                    }
                ],
                "travel_description": "12 x (Atlantis, 2 days, 6 travelers);"
            },
            {
                "selected_periods": "0c29d31a476861104a251bd8c26d4352,cc29d31a476861104a251bd8c26d4346,8029571a476861104a251bd8c26d4366",
                "duration_in_days": 4,
                "number_of_travelers": 2,
                "trip_location": "The Sun",
                "number_of_trips": 43,
                "sys_id": "1b0b1b5a476861104a251bd8c26d43c0",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 2000
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "2",
                            "periodUnit": "MONTH",
                            "optionOrder": "2",
                            "quantity": 3000
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "3",
                            "periodUnit": "WEEK",
                            "optionOrder": "3",
                            "quantity": 4000
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": [
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "4",
                        "periodUnit": "DAY",
                        "optionOrder": "4",
                        "quantity": 5000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "3",
                        "periodUnit": "WEEK",
                        "optionOrder": "3",
                        "quantity": 4000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "2",
                        "periodUnit": "MONTH",
                        "optionOrder": "2",
                        "quantity": 3000
                    },
                    {
                        "periodType": "BASE",
                        "periodUnitCount": "1",
                        "periodUnit": "YEAR",
                        "optionOrder": "1",
                        "quantity": 2000
                    }
                ],
                "travel_description": "43 x (The Sun, 4 days, 2 travelers);"
            },
            {
                "selected_periods": "0c29d31a476861104a251bd8c26d4352",
                "duration_in_days": 3,
                "number_of_travelers": 4,
                "trip_location": "Hogwarts",
                "number_of_trips": 3,
                "sys_id": "d30b5b5a476861104a251bd8c26d4377",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 2000
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": [
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "4",
                        "periodUnit": "DAY",
                        "optionOrder": "4",
                        "quantity": 5000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "3",
                        "periodUnit": "WEEK",
                        "optionOrder": "3",
                        "quantity": 4000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "2",
                        "periodUnit": "MONTH",
                        "optionOrder": "2",
                        "quantity": 3000
                    },
                    {
                        "periodType": "BASE",
                        "periodUnitCount": "1",
                        "periodUnit": "YEAR",
                        "optionOrder": "1",
                        "quantity": 2000
                    }
                ],
                "travel_description": "3 x (Hogwarts, 3 days, 4 travelers);"
            },
            {
                "selected_periods": "8029571a476861104a251bd8c26d4366",
                "duration_in_days": 10,
                "number_of_travelers": 8,
                "trip_location": "The Moon",
                "number_of_trips": 1,
                "sys_id": "d70b5b5a476861104a251bd8c26d435e",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "3",
                            "periodUnit": "WEEK",
                            "optionOrder": "3",
                            "quantity": 4000
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": [
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "4",
                        "periodUnit": "DAY",
                        "optionOrder": "4",
                        "quantity": 5000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "3",
                        "periodUnit": "WEEK",
                        "optionOrder": "3",
                        "quantity": 4000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "2",
                        "periodUnit": "MONTH",
                        "optionOrder": "2",
                        "quantity": 3000
                    },
                    {
                        "periodType": "BASE",
                        "periodUnitCount": "1",
                        "periodUnit": "YEAR",
                        "optionOrder": "1",
                        "quantity": 2000
                    }
                ],
                "travel_description": "1 x (The Moon, 10 days, 8 travelers);"
            },
            {
                "selected_periods": "8029571a476861104a251bd8c26d4366,cc29d31a476861104a251bd8c26d4346,0c29d31a476861104a251bd8c26d4352,c029571a476861104a251bd8c26d4342",
                "duration_in_days": 3,
                "number_of_travelers": 4,
                "trip_location": "Aruba",
                "number_of_trips": 3,
                "sys_id": "d70b9f5a476861104a251bd8c26d43b2",
                "instance_info": {
                    "selected_periods": [
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "3",
                            "periodUnit": "WEEK",
                            "optionOrder": "3",
                            "quantity": 4000
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "2",
                            "periodUnit": "MONTH",
                            "optionOrder": "2",
                            "quantity": 3000
                        },
                        {
                            "periodType": "BASE",
                            "periodUnitCount": "1",
                            "periodUnit": "YEAR",
                            "optionOrder": "1",
                            "quantity": 2000
                        },
                        {
                            "periodType": "OPTION",
                            "periodUnitCount": "4",
                            "periodUnit": "DAY",
                            "optionOrder": "4",
                            "quantity": 5000
                        }
                    ],
                    "need_for_entire_task_order_duration": false
                },
                "quantity_info_periods": [
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "4",
                        "periodUnit": "DAY",
                        "optionOrder": "4",
                        "quantity": 5000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "3",
                        "periodUnit": "WEEK",
                        "optionOrder": "3",
                        "quantity": 4000
                    },
                    {
                        "periodType": "OPTION",
                        "periodUnitCount": "2",
                        "periodUnit": "MONTH",
                        "optionOrder": "2",
                        "quantity": 3000
                    },
                    {
                        "periodType": "BASE",
                        "periodUnitCount": "1",
                        "periodUnit": "YEAR",
                        "optionOrder": "1",
                        "quantity": 2000
                    }
                ],
                "travel_description": "3 x (Aruba, 3 days, 4 travelers);"
            }
        ]
    }
}
```

image

image
image
image

## JSON structure of Training Estimate
```json
{
    "documentType": "INDEPENDENT_GOVERNMENT_COST_ESTIMATE",
    "templatePayload": {
        "training_estimates_test": [
            {
                "acquisition_package": "00b624f84764a1104a251bd8c26d43d8",
                "subscription_type": null,
                "training_estimated_values": "{\"80376cf84764a1104a251bd8c26d4306\":3,\"8037e8f84764a1104a251bd8c26d4311\":12}",
                "training_option": "MULTIPLE",
                "training_unit": "MONTHLY_SUBSCRIPTION",
                "estimated_price_per_training_unit": 120,
                "cloud_support_environment_instance": null,
                "sys_id": "0aa9ec7c4764a1104a251bd8c26d4347"
            },
            {
                "acquisition_package": "00b624f84764a1104a251bd8c26d43d8",
                "subscription_type": null,
                "training_estimated_values": "{}",
                "training_option": null,
                "training_unit": "ANNUAL_SUBSCRIPTION",
                "estimated_price_per_training_unit": 4004,
                "cloud_support_environment_instance": null,
                "sys_id": "2b5ca07047a4a1104a251bd8c26d431a"
            },
            {
                "acquisition_package": "00b624f84764a1104a251bd8c26d43d8",
                "subscription_type": null,
                "training_estimated_values": "{\"80376cf84764a1104a251bd8c26d4306\":20}",
                "training_option": "MULTIPLE",
                "training_unit": "PER_PERSON",
                "estimated_price_per_training_unit": 500,
                "cloud_support_environment_instance": null,
                "sys_id": "4f63cd8e47e021104a251bd8c26d436a"
            },
            {
                "acquisition_package": "00b624f84764a1104a251bd8c26d43d8",
                "subscription_type": null,
                "training_estimated_values": "{}",
                "training_option": null,
                "training_unit": "PER_CLASS",
                "estimated_price_per_training_unit": 700,
                "cloud_support_environment_instance": null,
                "sys_id": "76a9ec7c4764a1104a251bd8c26d433d"
            }
        ]
    }
}
```

### CODE REVIEW

**Comment by zachclark-ccpo** [comment]
image

**Comment by zachclark-ccpo** [comment]
image

Blocked by missing Cloud Support Environment Instance Id

**Comment by zachclark-ccpo** [comment]
Please be aware that there are 2 blockers for successfully running this DOCGEN:

- https://github.com/dod-ccpo/atat-web-ui/pull/1383
- A bug with the Unit Quantity Field on all IGCE Estimate Records (field not populating). @tonysumner-ccpo has been alerted of this bug yesterday evening.

**Comment by zachclark-ccpo** [comment]
Testing for only Travel and Training: ACQ0000416
```
/*


ACQ0000416 

1/26 Notes

Jedi Training
CONUS
Non-government facility
Location of CONUS training: Jedi Temple
4 personnel require this training
Need for entire TO: Yes
///
Per person
20 per person
5 people per period
05536c8b47e021104a251bd8c26d4379




SITH Training
Virtual Self Led
3 personnel require this training
Base and Option Period 1
30/ session
1 session base period, 3 sessions option period 3
f883ac8b47e021104a251bd8c26d438c


Magic Training
Virtual instructor-led
timezone: US East
3 personnel require this training
Base, op 1, op 2
annual
1000 per year



Security Plus Hundred
OCONUS
Location of OCONUS Training: Atlantis
3 personnel require this training
need for entire to duration: yes
///
30/month
5 months in all option perids, no base period


Zach Training
no preference
30 personnel require this training
need for entire to duration: yes
per session 20

20 sessions per period
ea4464cb47e021104a251bd8c26d431c




TRAVEL:
Canada
10 day, 10 people, 10 per period
Base Period only

France
20 days, 2 people, 1 per period
Base and Option period 1

Sword Coast:
4 days, 4 people, 4 per period
op 2, op 3

different price for base and each option period
base 100
op 1 200
op 2 300
op 3 400


*/
```

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 1]
```diff
@@ -177,9 +186,9 @@ Igce.prototype = {
 
                     // Insert into periods_estimate, under the specified period and period_line_items 
                     periods_estimate[indexOfCurrentPeriod].periodLineItems.push({
-                        "idiqClin": Igce.prototype.determineIdiqClin(item.classification_level, item.idiq_clin_type),// refactor to always return
+                        "idiqClin": Igce.prototype.determineIdiqClin(item.classification_level, item.idiq_clin_type),
                         "contractType": item.contract_type,
-                        "dowTaskNumber": item.dow_task_number,
+                        "dowTaskNumber": "TBD",//item.dow_task_number,
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:191]: Thumbs up

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 2]
```diff
@@ -254,6 +449,22 @@ Igce.prototype = {
             );
         }
     },
+    getTrainingEstimates: function (packageSysId) {
+        try {
+            var packageId = packageSysId ? packageSysId : this.acquisitionPackage.sys_id;
+            var limit = 50; // arbitrary
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:455]: @zachclark-ccpo Is this plenty large?

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 3]
```diff
@@ -238,6 +267,172 @@ Igce.prototype = {
...
+            case "PER_CLASS":
+                return "sessions";
+            case "MONTHLY_SUBSCRIPTION":
+                return "months";
+            default:
+                return "";
+        }
+    },
+
+    mapTrainingToPeriodsEstimate: function (training_data, periods_estimate) {
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:283]: Whew.  Nice effort determining what to do.

**Comment by jasonburkert-ccpo** [comment]
I assume these are still outstanding.  Approving but won't merge.

> Please be aware that there are 2 blockers for successfully running this DOCGEN:
> 
> * [bugfix/AT-8698-populate-training-estimate-with-cloud-support-environment-instance atat-web-ui#1383](https://github.com/dod-ccpo/atat-web-ui/pull/1383)
> * A bug with the Unit Quantity Field on all IGCE Estimate Records (field not populating). @tonysumner-ccpo has been alerted of this bug yesterday evening.

**Review by jasonburkert-ccpo — APPROVED** [review]

**Pull request merged**